authz/audit/stdout/stdout_logger.go

```diff
@@ -0,0 +1,101 @@
+/*
+ *
+ * Copyright 2023 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+// Package stdout contains a Stdout implementation of audit logging interface
+// The Stdoutlogger prints the audit.Event using log.go in json format
+package stdout
+
+import (
+	"encoding/json"
+	"log"
+	"time"
+
+	"google.golang.org/grpc/authz/audit"
+	"google.golang.org/grpc/grpclog"
+)
+
+var grpcLogger = grpclog.Component("authz-audit")
+
+type stdoutEvent struct {
+	FullMethodName string `json:"fullMethodName"`
+	Principal      string `json:"principal"`
+	PolicyName     string `json:"policyName"`
+	MatchedRule    string `json:"matchedRule"`
+	Authorized     bool   `json:"authorized"`
+	//Timestamp is populated using time.Now() during StdoutLogger.Log call
+	Timestamp string `json:"timestamp"`
+}
+
+// Stdout implementation of audit.Logger
+type StdoutLogger struct {
+}
+
+// Stdout implementation of audit.Logger.Log
+func (logger *StdoutLogger) Log(event *audit.Event) {
+	jsonBytes, err := json.Marshal(convertEvent(event))
+	if err != nil {
+		grpcLogger.Errorf("failed to marshal AuditEvent data to JSON: %v", err)
+	}
+	log.Println(string(jsonBytes))
+}
+
+const (
+	stdName = "stdout"
+)
+
+// Stdout implementation of audit.LoggerConfig
+// Since the logger doesn't support custom configs, it's a no-op one
+type StdoutLoggerConfig struct{}
+
+// No-op implementation of audit.LoggerConfig
+func (StdoutLoggerConfig) LoggerConfig() {}
```

Comment:
You shouldn't need to do this or add the export for `LoggerConfig`, the following should be sufficient in the `struct` declaration
```
type StdoutLoggerConfig struct {
	audit.LoggerConfig
}
```

See [embedding](https://go.dev/doc/effective_go#embedding) docs for details